libpam/pam_handlers.c

```diff
@@ -76,6 +78,23 @@ static int _pam_parse_conf_file(pam_handle_t *pamh, FILE *f
 	char **argv;
 	int argvlen;
 
+	if (x == (mod_buf_size - 1)) {
+	    mod_buf_size *= 2;
+	    buf = (char *) realloc(buf, mod_buf_size * sizeof(char));
+
+	    if (buf == NULL) {
+		pam_syslog(pamh, LOG_ERR,
+			   "Error Allocation Memory. Aborting! "
+		           "in file %s at line %d", __FILE__, __LINE__);
+                return PAM_ABORT;
+            }
+
+	    memset(buf, '\0', mod_buf_size * sizeof(char));
+            file_pos -= x;
+            fseek(f, file_pos, SEEK_SET);
```

Comment:
Sorry, but this is not acceptable, there shouldn't be any re-reading of the file.